workflow-to-js-transformer/src/test/java/com/github/cafdataprocessing/workflow/transform/WorkflowJavaScriptExecutionTest.java

```diff
@@ -67,6 +71,211 @@
+        try {
+            invocable.invokeFunction("processDocument", testDocument_1);
+            Assert.fail("Exception should have been thrown during processDocument call due to invalid script definition.");
+        }
+        catch(final ScriptException e) {
+            Assert.assertEquals(e.getCause().getCause().getClass(), RuntimeException.class, "Expecting script exception cause to be a " +
+                    "RuntimeException thrown due to invalid script definition.");
+        }
+    }
+
+    @Test(description = "Tests how a script that contains a single quote character on an action is handled during execution of " +
+            "workflow.")
+    public void addScriptWithSingleQuoteOnAction()
+            throws WorkerException, URISyntaxException, IOException, ApiException, WorkflowTransformerException,
+            ScriptException, NoSuchMethodException, DataStoreException {
+        final Map<String, Object> customDataInput = new HashMap<>();
+        final String testPropertyKey = "testProperty";
+        final String testPropertyValue = UUID.randomUUID().toString();
+        customDataInput.put(testPropertyKey, testPropertyValue);
+
+        final Map<String, Object> settingsInput = new HashMap<>();
+        settingsInput.put("customData", customDataInput);
+
+        // add an inline script
+        final String expectedScriptFieldValue_1 = UUID.randomUUID().toString();
+        final String testScriptValue_1 =
+                "function run(){document.getField('TEST').set('This < \\n > & '"+expectedScriptFieldValue_1+"');}";
+        final String testScriptName_1 = "testScript_1";
+        final Map<String, String> scriptEntry_1 = new HashMap<>();
+        scriptEntry_1.put("name", testScriptName_1);
+        scriptEntry_1.put("script", testScriptValue_1);
+
+        final List scriptsList = new ArrayList<>(Arrays.asList(scriptEntry_1));
+        settingsInput.put("scripts", scriptsList);
+
+        final FullAction testAction = BUILDER.buildFullAction("Script addition action",
+                new ArrayList<>(), 100, settingsInput);
+
+        final FullProcessingRule rule = BUILDER.buildFullProcessingRule("Test script addition rule", true, 100,
+                new ArrayList<>(Arrays.asList(testAction)),
+                new ArrayList<>());
+        final FullWorkflow workflow = BUILDER.buildFullWorkflow("Test script addition workflow", new ArrayList<>(Arrays.asList(rule)));
+
+        final Document testDocument_1 = DocumentBuilder.configure()
+                .withFields()
+                .addFieldValue("test", "string_value")
+                .documentBuilder().build();
+
+        try {
+            final Invocable invocable = getInvocableWorkflowJavaScriptFromFullWorkflow(workflow);
+            Assert.fail("This is expected to throw an exception until SCMOD-3725 is complete.");
+        }
+        catch(final ScriptException e) {
```

Comment:
Could we write the test like SCMOD-3725 is done and just add an @Ignore annotation to the test, that way all that needs to be done after SCMOD-3725 is the removal of the annotation.

Reply:
I'd rather we wait until SCMOD-3725 is done so we can be sure the test value being checked for is correct. I went to add it but got mixed up in how single quotes inside a single quoted string value should be handled and that kind of question is better handled under SCMOD-3725.